index.js

```diff
@@ -396,10 +406,27 @@ module.exports = class Hyperswarm extends EventEmitter {
     return cleared
   }
 
+  session (opts = {}) {
+    return new Hyperswarm({
+      ...this._opts,
+      seed: undefined,
+      keyPair: undefined,
+      ...opts,
+      dht: this.dht,
+      _root: this._root || this
+    })
+  }
+
   async destroy () {
     if (this.destroyed) return
```

Comment:
I wonder if we should return the destroy promise as well, so the loop with `session.destroy()` becomes more consistent?